### ISSUE TITLE
SOAPMessageContext.getMessage() on a fault mixes up namespaces

### ISSUE THREAD

**Issue by boskoop** [eclipse-ee4j/metro-jax-ws#58]
When calling `SOAPMessageContext.getMessage()` in a `SOAPHandler` the namespace of the fault detail ends up on the parent `detail` tag in the SOAP `Fault`.

I created a demo project to reproduce this issue. If in a SOAP call, a SOAP fault is created (given that `getMessage()` is not called in any handler) the response looks as follows:
```
<?xml version='1.0' encoding='UTF-8'?>
<S:Envelope xmlns:S="http://schemas.xmlsoap.org/soap/envelope/">
  <S:Body>
    <S:Fault xmlns:ns4="http://www.w3.org/2003/05/soap-envelope">
      <faultcode>S:Server</faultcode>
      <faultstring>Valar morghulis</faultstring>
      <detail>
        <HelloServiceException xmlns="http://xog.ch/jaxws/hello/v1">
          <ErrorText>Valar morghulis</ErrorText>
        </HelloServiceException>
      </detail>
    </S:Fault>
  </S:Body>
</S:Envelope>
```

If I call `SOAPMessageContext.getMessage()` in a server-side `SOAPHandler` (inside the `handleFault` method) the namespace of the payload exception is mixed up with the SOAP fault itself. Among other thins the namespace of `HelloServiceException` ends up on the `detail` tag:
```
<?xml version='1.0' encoding='UTF-8'?>
<S:Envelope xmlns:S="http://schemas.xmlsoap.org/soap/envelope/" xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">
  <SOAP-ENV:Header/>
  <S:Body xmlns="" xmlns:ns3="http://www.w3.org/2003/05/soap-envelope">
    <S:Fault>
      <faultcode>S:Server</faultcode>
      <faultstring>Valar morghulis</faultstring>
      <detail xmlns="http://xog.ch/jaxws/hello/v1">
        <HelloServiceException xmlns="http://xog.ch/jaxws/hello/v1">
          <ErrorText>Valar morghulis</ErrorText>
        </HelloServiceException>
      </detail>
    </S:Fault>
  </S:Body>
</S:Envelope>
```

This might be related to #39, but debugging showed that it did not go through the thereby mentioned code line, which is why I opened a new issue.

**Comment by boskoop** [comment]
I pushed the sample project on https://github.com/boskoop/metro-issue-58

**Comment by boskoop** [comment]
Error is demonstrated on the following test case:
`dontAskForAryaWithFaultLogging(ch.xog.jaxws.HelloServiceTest): Unexpected exception, expected<ch.xog.jaxws.hello.v1.HelloServiceException_Exception> but was<com.sun.xml.ws.fault.ServerSOAPFaultException>`

The mixup happens, when `smc.getMessage()` is called on line https://github.com/boskoop/metro-issue-58/blob/dcffecf94eb71d313b3435d883f04a008bcd8ff5/src/main/java/ch/xog/jaxws/handler/SOAPLoggingHandler.java#L53

**Comment by boskoop** [comment]
Hi,

Is there any news on this?

Thanks,
Christian

**Comment by boskoop** [comment]
I can confirm that this issue has been resolved on Metro 2.3.3+

See build result on the following versions:
Metro 2.3.3: https://github.com/boskoop/metro-issue-58/tree/metro-2.3.3
Metro 3.0.0-M4: https://github.com/boskoop/metro-issue-58/tree/master

For reference:
The test still fails on 2.3.2 (where the issue persists):
https://github.com/boskoop/metro-issue-58/tree/metro-2.3.2
